src/main.cpp

```diff
@@ -260,16 +271,63 @@ struct CNodeState {
     }
 };
 
-/** Map maintaining per-node state. Requires cs_main. */
```

Comment:
nit: You might want to at least make a comment about the expected locking order.